diff --git a/dist/index.js b/dist/index.js
index bd21e898f431298b70e941833c78819a1eb53a36..8ee782e75e1f71e025c217b791560d8125814dbb 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -4,6 +4,8 @@ function __insertCSS(code) {
   let head = document.head || document.getElementsByTagName('head')[0]
   let style = document.createElement('style')
   style.type = 'text/css'
+  const nonce = document.querySelector('meta[name="csp-nonce"]')?.content
+  if (nonce) style.nonce = nonce
   head.appendChild(style)
   ;style.styleSheet ? (style.styleSheet.cssText = code) : style.appendChild(document.createTextNode(code))
 }
diff --git a/dist/index.mjs b/dist/index.mjs
index d0c7a5a1fa1ed8b19c27bea968bb404f2437a07e..8db455762c27ea3c7e40bfe2f3a3ffdcc2590d38 100644
--- a/dist/index.mjs
+++ b/dist/index.mjs
@@ -4,6 +4,8 @@ function __insertCSS(code) {
   let head = document.head || document.getElementsByTagName('head')[0]
   let style = document.createElement('style')
   style.type = 'text/css'
+  const nonce = document.querySelector('meta[name="csp-nonce"]')?.content
+  if (nonce) style.nonce = nonce
   head.appendChild(style)
   ;style.styleSheet ? (style.styleSheet.cssText = code) : style.appendChild(document.createTextNode(code))
 }
